### PR TITLE
[StructuralMechanicsApp] Adding 2D 3 Nodes Timoshenko beam element

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
**📝 Description**
Continuation of https://github.com/KratosMultiphysics/Kratos/pull/12176

Here adding a 3 noded timoshenko element derived from the 2 noded one. I have override just a few methods to avoid duplication.

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_utilities/structural_mechanics_element_utilities.cpp [hunk 1]
```diff
@@ -250,6 +250,23 @@ double CalculateReferenceLength2D2N(const Element& rElement)
 /***********************************************************************************/
 /***********************************************************************************/
 
+double CalculateReferenceLength2D3N(const Element& rElement)
+{
+    KRATOS_TRY;
+
+    const array_1d<double, 3> delta_pos =
+        rElement.GetGeometry()[2].GetInitialPosition().Coordinates() -
+        rElement.GetGeometry()[0].GetInitialPosition().Coordinates();
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_utilities/structural_mechanics_element_utilities.cpp:259]: what if it is curved?

File: applications/StructuralMechanicsApplication/custom_elements/timoshenko_beam_element_2D3N.cpp [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+
+// Application includes
+#include "custom_elements/timoshenko_beam_element_2D3N.h"
+#include "custom_utilities/constitutive_law_utilities.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+void LinearTimoshenkoBeamElement2D3N::Initialize(const ProcessInfo& rCurrentProcessInfo)
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/timoshenko_beam_element_2D3N.cpp:27]: isnt this the same as in the baseclass?

File: applications/StructuralMechanicsApplication/custom_elements/timoshenko_beam_element_2D3N.cpp [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+    rN[5] = -L * Phi / (80.0 * Phi_square - 20.0 * Phi - 4.0) + 4.0 * L * xi_cube / (32.0 * Phi + 8.0) + 3.0 * L * xi_square / (160.0 * Phi_square - 40.0 * Phi - 8.0) - 2.0 * L * xi / (32.0 * Phi + 8.0) + 5.0 * xi_quad * (2.0 * L * Phi - L) / (160.0 * Phi_square - 40.0 * Phi - 8.0);
+    
+    rN *= 2.0 / L;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void LinearTimoshenkoBeamElement2D3N::GetSecondDerivativesShapeFunctionsValues(
+    VectorType& rN,
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/timoshenko_beam_element_2D3N.cpp:139]: perhaps a bounded type with compile time size (`BoundedVector`)?

**Review by philbucher — COMMENTED** [review]
nice!

what about tests?

**Comment by AlejandroCornejo** [comment]
> nice!
> 
> what about tests?

sure I'll add, this is a work in progress PR since there are a few details that we need to discuss

**Review by loumalouomega — COMMENTED** [review]
In general all Okay from my side

**Comment by AlejandroCornejo** [comment]
I'll create this PR later on again :)

**Pull request closed without merging**